synfig-studio/src/gui/modules/mod_palette/dock_paledit.cpp

```diff
@@ -349,6 +369,68 @@ Dock_PalEdit::add_color(const synfig::Color& x)
 	return size()-1;
 }
 
+void
+Dock_PalEdit::add_from_clipboard()
+{
+	if(clipboard_==NULL)
+		clipboard_ = gtk_clipboard_get(GDK_SELECTION_CLIPBOARD);
+	
+	gchar* hexcolor = gtk_clipboard_wait_for_text(clipboard_);
+
+	//Bool to check if the clipboard string is formatted in hexadecimal.  
+	bool hexfrmt = false;
+
+	//Check if clipboard is empty
+	if(hexcolor!=NULL){
+		hexfrmt = true;
+
+		//Check size is correct for the format: #ffffff
+		//Check if first character is a # for the correct format
+		if(strlen(hexcolor)!=7||hexcolor[0]!='#')
```

Comment:
There may also be a shortened version of the color representation.  (#rgb)
https://www.w3schools.com/css/css_colors_hex.asp

Reply:
I have added support for those additional hex representations 👍.